packages/bootstrap-chat/src/composed/ChatApp.jsx

```diff
@@ -75,39 +78,55 @@ function ChatApp({
   }
 
   function trackAgentName(agentName) {
-    if (agentName) {
-      const previouslyTrackedNames = taskContext.currentAgentNames || {};
-      const newAgentName = { [agentId]: agentName };
-      const currentAgentNames = { ...previouslyTrackedNames, ...newAgentName };
-      updateContext({ currentAgentNames: currentAgentNames });
-    }
+    const previouslyTrackedNames = taskContext.currentAgentNames || {};
+    const newAgentName = { [agentId]: agentName, agentName: agentName };
+    const currentAgentNames = { ...previouslyTrackedNames, ...newAgentName };
+    updateContext({ currentAgentNames: currentAgentNames });
   }
 
   let mephistoProps = useMephistoLiveTask({
-    onStateUpdate: ({ state, status }) => {
-      trackAgentName(state.agent_display_name);
-      if (state.task_done) {
-        setInputMode(INPUT_MODE.DONE);
```

Comment:
why don't we need this done case anymore?

Reply:
We still have it, just in `remaining_state.task_done` below. will be editing shortly though